packages/flutter/lib/src/services/raw_keyboard_fuschia.dart

```diff
@@ -0,0 +1,261 @@
+  ///
+  /// Dead keys are represented as Unicode combining characters.
+  final int codePoint;
+
+  /// The modifiers that were present when the key event occurred.
+  ///
+  /// See <https://fuchsia.googlesource.com/garnet/+/master/public/fidl/fuchsia.ui.input/input_event_constants.fidl>
+  /// for the numerical values of the modifiers. Many of these are also
+  /// replicated as static constants in this class.
+  ///
+  /// See also:
+  ///
+  ///  * [modifiersPressed], an accessor to get a Set of currently pressed
+  ///    modifiers.
+  ///  * [isModifierPressed], a function to query if a specific modifier is
+  ///    pressed.
+  ///  * [isControlPressed], a convenience accessor to see if the CTRL key is
+  ///    pressed.
+  ///  * [isShiftPressed], a convenience accessor to see if the SHIFT key is
+  ///    pressed.
+  ///  * [isAltPressed], a convenience accessor to see if the ALT key is
+  ///    pressed.
+  ///  * [isMetaPressed], a convenience accessor to see if the META key is
+  ///    pressed.
+  final int modifiers;
+
+  bool _isLeftRightModifierPressed(KeyboardSide side, int anyMask, int leftMask, int rightMask) {
+    if (modifiers & anyMask == 0) {
+      return false;
+    }
+    switch (side) {
+      case KeyboardSide.any:
+        return true;
+      case KeyboardSide.all:
+        return modifiers & leftMask != 0 && modifiers & rightMask != 0;
+      case KeyboardSide.left:
+        return modifiers & leftMask != 0;
+      case KeyboardSide.right:
+        return modifiers & rightMask != 0;
+    }
+    return false;
+  }
+
+  @override
+  bool isModifierPressed(ModifierKey key, {KeyboardSide side = KeyboardSide.any}) {
+    assert(side != null);
+    switch (key) {
+      case ModifierKey.controlModifier:
+        return _isLeftRightModifierPressed(side, modifierControl, modifierLeftControl, modifierRightControl);
+      case ModifierKey.shiftModifier:
+        return _isLeftRightModifierPressed(side, modifierShift, modifierLeftShift, modifierRightShift);
+      case ModifierKey.altModifier:
+        return _isLeftRightModifierPressed(side, modifierAlt, modifierLeftAlt, modifierRightAlt);
+      case ModifierKey.metaModifier:
+        return _isLeftRightModifierPressed(side, modifierMeta, modifierLeftMeta, modifierRightMeta);
+      case ModifierKey.capsLockModifier:
+        return modifiers & modifierCapsLock != 0;
+      case ModifierKey.numLockModifier:
+      case ModifierKey.scrollLockModifier:
+      case ModifierKey.functionModifier:
+      case ModifierKey.symbolModifier:
+        // Fuschia doesn't have masks for these keys (yet).
+        return false;
+    }
+    return false;
+  }
+
+  @override
+  KeyboardSide getModifierSide(ModifierKey key) {
+    KeyboardSide findSide(int leftMask, int rightMask, int combinedMask) {
+      KeyboardSide side;
+      final int combined = modifiers & combinedMask;
+      if (combined == leftMask) {
+        side = KeyboardSide.left;
```

Comment:
Yeah, I did look for that, but it just seemed best to implement them separately, because there were subtle differences that made sharing less attractive.

They're not exactly the same: Fuchsia masks are more sensible (both == left | right), so I don't have to compute that ahead of time, whereas on Android, they have three bits, one for left, right, and 'any'.

And because Fuchsia doesn't support some modifiers yet, the handling is different for the case too, so the main body can't be shared.